### PR TITLE
Draft,  Move OpenSSL Android build to the target directory #8568

### LINKED ISSUE #8568: Move OpenSSL Android build to the target directory
Right now, OpenSSL is downloaded and built for Android in the support/android directory instead of in the target/arm-linux-androideabi/FLAVOR directory. See:
https://github.com/servo/servo/blob/master/support/android/openssl.makefile

This task would involve having our python code that calls the makefile set or pass in the proper target directory location:
https://github.com/servo/servo/blob/d16ba51b4722a84f69976ca8679af672495248c8/python/servo/build_commands.py#L219

Then, the new makefile would instead download the .tar.gz file into that location, unpack it there, and also build it there.

For this issue, you will need to be able to build for Android: https://github.com/servo/servo/wiki/Building-for-Android

### PR DESCRIPTION
**Pull request by mskrzypkows**
I've created a draft, please give me feedback. Is it good direction?

Fixes #8568

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8626)

### CODE REVIEW

File: python/servo/build_commands.py [hunk 1]
```diff
@@ -220,14 +220,19 @@ def build(self, target=None, release=False, dev=False, jobs=None,
             make_cmd = ["make"]
             if jobs is not None:
                 make_cmd += ["-j" + jobs]
-            with cd(self.android_support_dir()):
+            android_path =  path.join(self.get_target_dir(), "arm-linux-androideabi")
+            if dev:
+                android_path =  path.join(android_path, "debug")
+            else:
+                android_path =  path.join(android_path, "release")
```
Comment (frewsxcv) [inline python/servo/build_commands.py]: Alternatively:

```python
android_path = path.join(android_path, "debug" if dev else "release")
```

**Comment by larsbergstrom** [comment]
If it works (see the instructions for building on Android in the original issue), assuming that @frewsxcv's comment is addressed, I'd be more than happy to r+ this!

**Comment by mskrzypkows** [comment]
OK, I'll push next commit.

**Comment by larsbergstrom** [comment]
@mskrzypkows This looks great! Did it successfully build for Android? If you need help testing it, I can do so (or we can kick off a `try` build, though not until we fix some infrastructure problems we're having this morning).

**Comment by mskrzypkows** [comment]
It seems not be merged, please retray.

**Pull request merged**